articles/django-app-using-postgresql-database/index.md

```diff
@@ -0,0 +1,313 @@
+
+Let's install `django` and `psycopg2` using `pip`. `psycopg2` is a popular PostgreSQL database adapter for Python.
+
+```bash
+pip install django psycopg2
+```
+  
+Let's create our Hello World Django project.
+
+```bash
+ django-admin startproject django_app
+```
+
+Change your current directory into our Django project `django_app`.
+
+```bash
+cd django_app
+```
+
+Let's create an app in our `django_app` project.
+
+> A Django application is a Python package that is specifically intended for use in a Django project. Django apps are reusable in different Django project. [Django](https://docs.djangoproject.com/en/3.1/ref/applications/)
+
+```bash 
+django-admin startapp hello_world
+```
+
+For our `hello_world` app to work we need to register the app in the `django_app` project. To do this, let's navigate to the `django_app/settings.py`.
+
+Add the `hello_world` app name to the installed apps section of the file.
+
+```python 
+...
+INSTALLED_APPS = [
+    ...
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
+    'hello_world',  # registering the hello_world app
+]
+```
+
+### Step 3: Set up your project to use PostgreSQL Database.
+SQlite is the default database that comes with Django. We need the to change the database configurations to use PostgreSQL.
+
+Let's navigate to `django_app/settings.py`. In the `DATABASES` section, you'll see the code snippets below.
+
+```python 
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.db.backends.sqlite3',
+        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
+    }
+}
+```
+
+Change the above code snippet to this:
+
+```python 
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.db.backends.postgresql_psycopg2',
+        'NAME': 'helloworld',
+        'USER': '<yourname>',
+        'PASSWORD': 'password',
+        'HOST': 'localhost',
+        'PORT': '',
+    }
+}
+```
+
+We changed the backend engine to use `postgresql_psycopg2` instead of the default `sqlite3`.
+
+- NAME is the name of the database we created for our project.
+
+- USER is the database user we created during the database creation.
+
+- PASSWORD is the password to the database we created.
+
+Now, let's make sure that we have connected to the Postgres database.
+
+> The **manage.py** file is a python script which allows us to perform adminstartive tasks like migrations, running the server, etc.
+
+```python
+python manage.py makemigrations
+```
+
+The `makemigrations` command is responsible for creating new migrations based on the changes you have made to our models. Since we don't have a model in our Django project, we will not worry much about this. To get a better understanding of what this command does, check out this article [Django Migrations: A Primer](https://realpython.com/django-migrations-a-primer/).
+
+When you run it for the first time, the default models in the Django project will get migrated.
```

Comment:
Please mention that the `makemigrations` command created a file with all the SQL commands that will be run while running the `migrate` command. This is the most important part of the `makemigrations` command.